posttroll/tests/test_pubsub.py

```diff
@@ -627,6 +627,52 @@ def test_dict_config_full_subscriber(Subscriber_update):
     _ = create_subscriber_from_dict_config(settings)
 
 
+def test_publisher_tcp_keepalive():
+    """Test that TCP Keepalive is set for Publisher if the environment variables are present."""
+    _set_tcp_keepalive_env_variables()
+
+    socket = mock.MagicMock()
+    with mock.patch('posttroll.get_context') as get_context:
+        get_context.return_value.socket.return_value = socket
+        from posttroll.publisher import Publisher
+
+        _ = Publisher("tcp://127.0.0.1:9000")
+
+        _test_tcp_keepalive(socket)
+
+
+def _set_tcp_keepalive_env_variables():
+    import os
+
+    os.environ["POSTTROLL_TCP_KEEPALIVE"] = "1"
+    os.environ["POSTTROLL_TCP_KEEPALIVE_CNT"] = "10"
+    os.environ["POSTTROLL_TCP_KEEPALIVE_IDLE"] = "1"
+    os.environ["POSTTROLL_TCP_KEEPALIVE_INTVL"] = "1"
```

Comment:
To make sure the environment is reverted for other tests, maybe we could use `monkeypatch`? https://docs.pytest.org/en/stable/how-to/monkeypatch.html#monkeypatching-environment-variables

Reply:
Thanks! I've been hitting my head on the wall for the past hour or so. Not that the tests pass even with the latest commits (I tried with a class and locking in between). If I run only my new tests, they work great, so guess I need to fix the other tests that use `os.environ` too?

Reply:
Depending on what they set, yeah probably.

Reply:
It's these tests themselves that cause the problems, even with the monkeypatched fixture. If I move these two tests with the necessary stuff to another Python file, the tests fail. If I run them individually, they pass.

Reply:
Adding `print(os.environ['POSTTROLL_TCP_KEEPALIVE'])` to either of the tests raise a `KeyError`, so the env variables were not set by the monkeypatch fixture.

Reply:
wtf?

Reply:
Ok, now the tests should work. I was mocking the *original* location of `get_context()`, not where it is used.